File: OCH_Tao/tessoku-book/B07.js
const input = require("fs").readFileSync("/dev/stdin", "utf8").trim().split("\n");
const T = parseInt(input.shift(), 10);
input.shift();
let X = Array(T + 1).fill(0);
for (const i of input) {
  const [L, R] = i.split(" ").map(x => parseInt(x, 10));
  X[L]++;
  X[R]--;
}
let ans = [0];
for (let i = 0; i < T; i++) {
  ans.push(ans[i] + X[i]);
}
for (let i = 1; i < T + 1; i++) {
  console.log(ans[i]);
}

File: OCH_Tao/tessoku-book/B07.py
from itertools import accumulate
T = int(input())
N = int(input())
X = [0]*(T+1)
for i in range(N):
  L, R = map(int, input().split())
  X[L] += 1
  X[R] -= 1
ans = list(accumulate([0]+X))[1:]
[print(ans[i]) for i in range(T)]
